docs/meshstack.openstack.index.md

```diff
@@ -48,3 +48,36 @@ Project Users get the following OpenStack roles:
 The actual access rights associated with these roles are managed by the OpenStack instance and are not part of meshStack.
 
 More details about the User Federation with OpenStack can be found [here](meshstack.openstack.index.md).
+
+## Federated Idp
+
+meshIdB is not required for the integration anymore. You can directly integrate your company-wide IdP to OpenStack. To do this its necessary to enable the federatedIdp within the OpenStack plaform configuration in meshfed. Besides that the Idp needs to be fully integrated within OpenStack. This means users should have access to OpenStack independent of the panel login. During the automated replication meshStack will make sure that users have access to the OpenStack projects they are assigned to in meshStack. Every new created User in OpenStack will be assigned to the MeshUsers domain to prevent the creation of one user multiple times. If the user still exists in OpenStack then the automatisation will pick this user to assign him to the specific groups. We create for each project an own group.
```

Comment:
```suggestion
You can directly integrate your company-wide IdP to OpenStack. This simplifies the integration with meshStack compared to the meshIdB integration that is explained above. Using the company-wide IdP must be enabled in meshStack for each affected OpenStack platform.
```
at this place you should show how to configure it. The cleanest solution for it would be actual documentation snippets in dhall. But as we don't have that for the OpenStack configuration at all till now, it would be a bit out of scope for now as you would have to describe every config option in the OpenStack platform config. So i'd recommend to use the snippet you added below, but reduce it to only the relevant config attributes. Just mention those that are relevant for the federated IdP configuration. That are federated-idp-enabled, idp-provider, idp-protocol.

```suggestion
In OpenStack you have to configure the company-wide IdP as described here (link to OpenStack docs, ask @henryde). Make sure to configure the MeshUsers domain as the domain of all users coming from the IdP like this (get the example from @henryde).

During replication, meshStack will make sure that users have access to the OpenStack projects they are assigned to in meshStack. Users will be assigned to the according groups (per meshProject) meshStack creates in OpenStack. If 
meshStack is about to assign a user that does not exist in OpenStack yet, meshStack will create this user with the according federatedIdP attributes and create the user in `MeshUsers` domain. That way once the user logs in via the IdP, he will be mapped to the user created by meshStack. If the user already exists in OpenStack during replication, meshStack will pick up this user to assign him to the according groups.
```

Reply:
All information are necessary so I'll leave the example like it is.
MeshUsers are not "Henry" related. We'll create the MeshUsers domain during the ReplicateFederatedUsersStep. So I don't aggree to add an example.
I've asked henry about the os config. I've added the docs link.

Reply:
I actually do think that they are "Henry" related, as Henry had to configure something in OpenStack, so users are actually put into the MeshUsers domain when they log in to OpenStack. Otherwise they were alway created in the default domain and not mapped to the users we created

Reply:
And regarding the dhall config ... the problem i see with mentioning all attributes is, that it is mentione din teh federatedIdP section of the docs and e.g. "default-added-roles" or the credentials are not related to that topic at all. So if you mention them here, you also have to explain what they are actually used for. But that doesn't belong to the federated IdP section and therefore doesn't belong here.

The cleanest way would be to have a complete integration section that describe the complete platform configuration with all attributes. That one would use our dhall snippet approach to cleanly document our dhall config. And then you would only link to that section from the federated IdP section and mention that the 3 attributes mentioned before are relevant for the federated IdP config. But that change is out of scope for this PR imo and therefore i'd go with just mentioning the 3 attributes that are actually related to federatedIdP config.

Reply:
Ok. I'll remove the other lines.